WFLY-16453 Create a layer for singleton subsystem

rhusar opened the pull request:
Resolves
https://issues.redhat.com/browse/WFLY-16453

Required by https://github.com/wildfly/wildfly/pull/15609

darranl commented:
@rhusar do you think the test failures on the preview job are related?

rhusar commented:
@darranl Yes, those were related (and related to https://issues.redhat.com/browse/WFLY-16580). I have already pushed up a fix for that particular failure.

pferraro commented on testsuite/integration/clustering/pom.xml:
```diff
@@ -1537,6 +1654,44 @@
+                                    <record-state>false</record-state>
+                                    <log-time>${galleon.log.time}</log-time>
+                                    <offline>true</offline>
+                                    <plugin-options>
+                                        <jboss-maven-dist/>
+                                        <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                                        <optional-packages>passive+</optional-packages>
+                                    </plugin-options>
```
N.B. This will impact https://github.com/wildfly/wildfly/pull/15730

rhusar replied:
Right, thanks, that's painfully clear. :-)

pferraro commented on testsuite/integration/clustering/pom.xml:
```diff
@@ -1499,6 +1555,67 @@
+                                    <record-state>false</record-state>
+                                    <log-time>${galleon.log.time}</log-time>
+                                    <offline>${galleon.offline}</offline>
+                                    <plugin-options>
+                                        <jboss-maven-dist/>
+                                        <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                                        <optional-packages>passive+</optional-packages>
+                                    </plugin-options>
```
N.B. This will impact https://github.com/wildfly/wildfly/pull/15730

rhusar replied:
👍

rhusar commented on testsuite/integration/clustering/pom.xml:
```diff
@@ -1161,8 +1162,45 @@
+                                        <jboss.server.config.file.name>standalone-full-ha.xml</jboss.server.config.file.name>
+                                        <!-- Override ${server.jvm.arg} to only include ${jvm.args.ip} but *not* include ${jvm.args.ip.server} -->
+                                        <server.jvm.args>${surefire.system.args} ${jvm.args.ip} ${jvm.args.other} ${jvm.args.timeouts} ${jvm.args.dirs} ${extra.server.jvm.args} -Dnode0=${node0} -Dnode1=${node1} -Dnode2=${node2} -Dnode3=${node3}</server.jvm.args>
+                                        <infinispan.server.home>${infinispan.server.home.template}</infinispan.server.home>
+                                        <infinispan.server.profile>${infinispan.server.profile.override}</infinispan.server.profile>
```
@pferraro Yes, at the moment they are required by `org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase`'s `InfinispanServerRuleBuilder`. Coincidentally, I am fixing it in a different branch.

bstansberry reviewed (changes requested):
Please add executions for these two layers to LayersTestCase.

We should also add one to the 'all-layers-provisioning-full' execution and the other to the 'all-layers-jpa-distributed-provisioning-full' (and rename the later to remove 'jpa' as it's been a while since it was only about JPA).

rhusar commented:
@bstansberry I have actually noticed the missing coverage already and have it in fixed a branch (will push shortly), thanks!

rhusar committed:
WFLY-16453 Create layers for singleton subsystem

rhusar commented:
I have reworked the layers a bit to actually depend on the "base-server" so that they can be used easily with other layers; the -ha variant also needs to depend on transactions; tests were also updated to cover all this. So re-requesting review. Thanks!

rhusar commented:
/retest

bstansberry commented:
Thanks @rhusar
